### PR TITLE
Bump PocoCpp to 1.7.8

### PR DESCRIPTION
**Pull request by shekharhimanshu**
- Add PocoCpp versions, [1.7.5](https://github.com/pocoproject/poco/releases/tag/poco-1.7.5-release) through [1.7.8](https://github.com/pocoproject/poco/releases/tag/poco-1.7.8-release)
- Bump PocoCpp default to [1.7.8](https://github.com/pocoproject/poco/releases/tag/poco-1.7.8-release)

### CODE REVIEW

**Comment by ruslo** [comment]
Poco use `OpenSSL` and should be hunterized, see https://github.com/hunter-packages/poco/commit/6ae25e2b94c641ba65291ce4b2eb0d21a1566c34

You should send updates to https://github.com/hunter-packages/poco/ first.

**Comment by shekharhimanshu** [comment]
Could you please explain, `Poco use OpenSSL and should be hunterized`?
IMHO, projects using Poco should resolve OpenSSL (and other) dependencies themselves instead of being forced to download and build those dependencies using Hunter.

Since Poco uses OpenSSL, here's an example of how Poco can be built on OSX, 
Hunter's `config.cmake`:
```
# OpenSSL must be downloaded manually beforehand using, "brew install openssl"
execute_process(COMMAND brew --prefix openssl OUTPUT_VARIABLE BREW_OPENSSL_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)

# Pass OpenSSL find paths to Hunter
set(HUNTER_CMAKE_OPENSSL_ARGS
        "OPENSSL_ROOT_DIR=${BREW_OPENSSL_DIR}"
        "OPENSSL_INCLUDE_DIR=${BREW_OPENSSL_DIR}/include"
)

# customize Poco
hunter_config(
        PocoCpp
        VERSION
        ${POCO_VERSION}
        CMAKE_ARGS
        ${HUNTER_CMAKE_OPENSSL_ARGS}
        POCO_STATIC=ON
        POCO_UNBUNDLED=OFF
        ENABLE_XML=ON
        ENABLE_JSON=ON
        ENABLE_UTIL=ON
        ENABLE_NET=ON
        ENABLE_NETSSL=ON
        ENABLE_CRYPTO=ON
        ENABLE_DATA=ON
        ENABLE_DATA_SQLITE=ON
        ENABLE_ZIP=ON
        ENABLE_TESTS=OFF
)
```

This way, projects building Poco with Hunter can decide for themselves how they want to get OpenSSL . For example, in my case I used `homebrew` to install OpenSSL on mac.

Poco uses a lot of other libraries besides OpenSSL like, `APR`, `Apache2`, `EXPAT`, `SQLite3` to name a few (depending upon how its build), will you propose in that case, to have them all built using Hunter as well?

**Comment by headupinclouds** [comment]
> IMHO, projects using Poco should resolve OpenSSL (and other) dependencies themselves instead of being forced to download and build those dependencies using Hunter.

Hunter is designed around the concept of building redistributable SW from source from a consistent toolchain for predictable results.  FWIW, this has come up before, and I recall seeing some notes about mixing packages managers (i.e., homebrew + hunter) here:

https://docs.hunter.sh/en/latest/faq/interaction-with-other-package-managers.html

Which mentions an open issue/request for including system packages here:

https://github.com/ruslo/hunter/issues/501

Using system packages does seem to introduce a lot of complexity and uncertainty wrt ABI compatibility, etc.

You can always disable hunter `HUNTER_ENABLED=OFF` to default to system packages (if available) in your current `find_package()` calls, although that doesn't seem to be what you want here.

**Comment by ruslo** [comment]
> This way, projects building Poco with Hunter can decide for themselves how they want to get OpenSSL . For example, in my case I used homebrew to install OpenSSL on mac.

And if I want to use POCO **and** OpenSSL from Hunter in your project you will have a conflict:
* https://docs.hunter.sh/en/latest/creating-new/cmake-dependencies.html#conflict

> Poco uses a lot of other libraries besides OpenSSL like, APR, Apache2, EXPAT, SQLite3 to name a few (depending upon how its built), will you propose in that case, to have them all built using Hunter as well?

Some of them is not present in Hunter yet, so as a **workaround** the version from system can be used. But in general yes, they all should be managed by Hunter.

**Comment by shekharhimanshu** [comment]
Thank you both for the explanation. I understand your concern regarding using system packages especially in case of a conflict.

>Some of them is not present in Hunter yet, so as a workaround the version from system can be used. But in general yes, they all should be managed by Hunter.

In that case, ideally I would like to refactor `hunterized` Poco so that all the dependencies are managed by Hunter itself but that seems like a lot of work at the moment and out of scope of this PR.
I might summit another PR for that when I get time.

For now, I will just update the hunterized Poco to latest [develop](https://github.com/pocoproject/poco/tree/develop) and submit a separate PR.

**Comment by ruslo** [comment]
> so that all the dependencies are managed by Hunter itself but that seems like a lot of work at the moment

@shekharhimanshu you don't have to add everything now, the OpenSSL was added because somebody needed it. So the issue with current pull request is that we will break this stuff hence it's not acceptable.

> For now, I will just update the hunterized Poco to latest develop and submit a separate PR

Tell me what version you need, I will create branch `hunter-X.Y.Z` in https://github.com/hunter-packages/poco based on `X.Y.Z` upstream tag. Then you can send there hunterization patches from previous https://github.com/hunter-packages/poco/releases/tag/v1.7.6-p0 version.

**Comment by shekharhimanshu** [comment]
Yeah sure. I need [1.7.8](https://github.com/pocoproject/poco/releases/tag/poco-1.7.8-release)

**Comment by ruslo** [comment]
https://github.com/hunter-packages/poco/tree/hunter-1.7.8

**Comment by ruslo** [comment]
Cherry-pick top 3 commit from `hunter` should do the trick:
* https://github.com/hunter-packages/poco/commits/hunter

**Comment by shekharhimanshu** [comment]
Hi, sorry for the delay.

Seems like I don't have write permission to `hunter-packages/poco`?

**Comment by ruslo** [comment]
> Seems like I don't have write permission to hunter-packages/poco?

Check your email.

**Comment by shekharhimanshu** [comment]
Updated.

**Comment by ruslo** [comment]
Please squash everything into one commit.

**Comment by shekharhimanshu** [comment]
Squashed.

File: cmake/projects/PocoCpp/hunter.cmake [hunk 1]
```diff
@@ -8,6 +8,17 @@ hunter_add_version(
     PACKAGE_NAME
     PocoCpp
     VERSION
+    v1.7.8-p0
```
Comment (ruslo) [inline cmake/projects/PocoCpp/hunter.cmake]: Should be just `1.7.8-p0`, not `v1.7.8-p0`. This form used for most of versions.

Comment (shekharhimanshu) [inline cmake/projects/PocoCpp/hunter.cmake]: My bad.
Corrected and squashed again.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2120
* https://travis-ci.org/ingenue/hunter/builds/262242092?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2123
* https://travis-ci.org/ingenue/hunter/builds/262292384?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.62

**Pull request merged**